velox/dwio/common/SelectiveColumnReader.h

```diff
@@ -151,6 +151,19 @@ class SelectiveColumnReader {
   // from a downstream operator.
   virtual void resetFilterCaches();
 
+  virtual void resetWhenFilterAll() {
+    rawStringBuffer_ = nullptr;
+    rawStringSize_ = 0;
+    rawStringUsed_ = 0;
+    for (auto& buff : stringBuffers_) {
+      buff->release();
+    }
+    stringBuffers_.clear();
```

Comment:
Instead of throwing all the buffers away, can we keep and reuse the largest one?